checker/src/main/java/alluxio/checker/SparkIntegrationChecker.java

```diff
@@ -0,0 +1,243 @@
+    FAILCLASS, // Spark driver or executors cannot recognize Alluxio classes
+    FAILFS,    // Spark driver or executors cannot recognize Alluxio filesystem
+    FAILHA,    // Spark driver cannot support Alluxio-HA mode
+    SUCCESS;
+  }
+
+  /**
+   * Implements check integration of Spark with Alluxio.
+   *
+   * @param sc current JavaSparkContext
+   * @return Status performIntegrationChecks results
+   */
+  private Status run(JavaSparkContext sc) {
+    // Checks whether Spark driver can recognize Alluxio classes and filesystem
+    Status driverStatus = performIntegrationChecks();
+    if (driverStatus.equals(Status.FAILCLASS)) {
+      LOG.error("Spark driver: " + getAddress() + " failed to recognize Alluxio classes.");
+      return driverStatus;
+    } else if (driverStatus.equals(Status.FAILFS)) {
+      LOG.error("Spark driver: " + getAddress() + " failed to recognize Alluxio filesystem.");
+      return driverStatus;
+    } else {
+      LOG.info("Spark driver: " + getAddress() + " can recognize Alluxio filesystem.");
+    }
+
+    // Supports Alluxio high availability mode
+    if (alluxio.Configuration.getBoolean(PropertyKey.ZOOKEEPER_ENABLED)) {
+      mAlluxioHAMode = true;
+      if (!alluxio.Configuration.containsKey(PropertyKey.ZOOKEEPER_ADDRESS)) {
+        LOG.error("Zookeeper address has not been set:");
+        return Status.FAILHA;
+      }
+    }
+
+    return runSparkJob(sc);
+  }
+
+  /**
+   * Spark job to check whether Spark executors can recognize Alluxio filesystem.
+   *
+   * @param sc current JavaSparkContext
+   * @return Status
+   */
+  private Status runSparkJob(JavaSparkContext sc) {
+    // Generates a list of integer for testing
+    List<Integer> nums = IntStream.rangeClosed(1, mPartitions).boxed().collect(Collectors.toList());
+
+    JavaRDD<Integer> dataSet = sc.parallelize(nums, mPartitions);
+
+    // Runs a Spark job to check whether Spark executors can recognize Alluxio
+    JavaPairRDD<Status, String> extractedStatus = dataSet
+        .mapToPair(s -> new Tuple2<>(performIntegrationChecks(), getAddress()));
+
+    // Merges the IP addresses that can/cannot recognize Alluxio
+    JavaPairRDD<Status, String> mergeStatus = extractedStatus.reduceByKey((a, b)
+        -> a.contains(b) ?  a : (b.contains(a) ? b : a + "; " + b));
+
+    mSparkJobResult = mergeStatus.collect();
+
+    // If one executor cannot recognize Alluxio, something wrong has happened
+    boolean canRecognizeClass = true;
+    boolean canRecognizeFS = true;
+    for (Tuple2<Status, String> op : mSparkJobResult) {
+      if (op._1().equals(Status.FAILCLASS)) {
+        canRecognizeClass = false;
+      }
+      if (op._1().equals(Status.FAILFS)) {
+        canRecognizeFS = false;
+      }
+    }
+    return canRecognizeClass ? (canRecognizeFS ? Status.SUCCESS : Status.FAILFS) : Status.FAILCLASS;
+  }
+
+  /**
+   * Checks if this Spark driver or executors can recognize Alluxio classes and filesystem.
+   *
+   * @return Status
```

Comment:
Our code base convention: for a method takes no input arg, the javadoc can be as simple as:
```
/**
 * @return if this Spark driver or executors can recognize Alluxio classes and filesystem
 */
```